modules/client/src/main/java/org/apache/ignite/internal/client/compute/ClientCompute.java

```diff
@@ -140,22 +143,42 @@ public <T, R> CompletableFuture<BroadcastExecution<R>> submitAsync(
             AllNodesBroadcastJobTarget allNodesBroadcastTarget = (AllNodesBroadcastJobTarget) target;
             Set<ClusterNode> nodes = allNodesBroadcastTarget.nodes();
 
+            //noinspection unchecked
             CompletableFuture<SubmitResult>[] futures = nodes.stream()
                     .map(node -> executeOnAnyNodeAsync(Set.of(node), descriptor, arg))
                     .toArray(CompletableFuture[]::new);
 
-            // Wait for all the futures but don't fail resulting future, keep individual futures in executions.
-            return allOf(futures).handle((unused, throwable) -> new BroadcastJobExecutionImpl<>(
-                    Arrays.stream(futures)
-                            .map(fut -> mapSubmitResult(descriptor, cancellationToken, fut))
-                            .collect(Collectors.toList())
-            ));
+            return mapSubmitFutures(futures, descriptor, cancellationToken);
+        } else if (target instanceof TableJobTarget) {
+            TableJobTarget tableJobTarget = (TableJobTarget) target;
+            String tableName = tableJobTarget.tableName();
+            return getTable(tableName)
+                    .thenCompose(table -> table.partitionManager().primaryReplicasAsync()
+                            .thenCompose(replicas -> {
```

Comment:
```suggestion
                    .thenCompose(table -> table.partitionManager().primaryReplicasAsync())
                    .thenCompose(replicas -> {
```

A bit easier to comprehend, IMO.